.github/workflows/publish.yml

```diff
@@ -0,0 +1,37 @@
+name: NPM Publish
+
+on:
+  workflow_dispatch:
```

Comment:
What I use on other projects (for next PRs):
- PR labels decide which semantic versioning the change represent (patch, minor, major)
- PR merge creates or update a draft github release
- Publishing the release pushes the new version to NPM

Examples on https://github.com/castore-dev/castore:
- https://github.com/castore-dev/castore/blob/main/.github/workflows/draft-or-update-next-release.yml
- https://github.com/castore-dev/castore/blob/main/.github/workflows/release-to-npm.yml

Reply:
Great idea!